qcodes_contrib_drivers/drivers/NationalInstruments/RFSG.py

```diff
@@ -0,0 +1,245 @@
+    PXI-5610,  PXI-5650,  PXI-5651,  PXI-5652,  PXI-5670,  PXI-5671, PXIe-5611,
+    PXIe-5644, PXIe-5645, PXIe-5646, PXIe-5650, PXIe-5651, PXIe-5652,
+    PXIe-5653, PXIe-5654, PXIe-5672, PXIe-5673, PXIe-5673E, PXIe-5820,
+    PXIe-5840.
+
+    Documentation for the NI-RFSG C API can be found by default in the
+    folder C:\Users\Public\Documents\National Instruments\NI-RFSG\Documentation
+
+    Only very basic functionality is implemented.
+
+    Tested with
+
+    - PXIe-5654
+
+    Args:
+        name: Name for this instrument
+        resource: Identifier for this instrument in NI MAX.
+        dll_path: path to the NI-RFSG library DLL. If not provided, use the
+            default location,
+            ``C:\Program Files\IVI Foundation\IVI\bin\NiRFSG_64.dll``.
+        id_query: whether to perform an ID query on initialization
+        reset_device: whether to reset the device on initialization
+    """
+
+    # default DLL location
+    dll_path = r"C:\Program Files\IVI Foundation\IVI\bin\NiRFSG_64.dll"
+    # C:\Program Files (x86)\IVI Foundation\IVI\bin\NiRFSG.dll for 32-bit
+
+    def __init__(self, name: str, resource: str,
+                 dll_path: Optional[str] = None,
+                 id_query: bool = False,
+                 reset_device: bool = False,
+                 **kwargs):
+
+        super().__init__(name=name, resource=resource,
+                         dll_path=dll_path or self.dll_path,
+                         lib_prefix="niRFSG", **kwargs)
+
+        # Wrap DLL calls
+        self.wrapper.Initiate = self.wrapper.wrap_dll_function_checked(
+                name_in_library="Initiate",
+                argtypes=[NamedArgType("vi", ViSession)]
+                )
+
+        self.wrapper.Abort = self.wrapper.wrap_dll_function_checked(
+                name_in_library="Abort",
+                argtypes=[NamedArgType("vi", ViSession)])
+
+        self.wrapper.ConfigureRF = self.wrapper.wrap_dll_function_checked(
+                name_in_library="ConfigureRF",
+                argtypes=[
+                    NamedArgType("vi", ViSession),
+                    NamedArgType("frequency", ViReal64),
+                    NamedArgType("powerLevel", ViReal64),
+                    ]
+                )
+
+        self.add_parameter(name="frequency",
+                           unit="Hz",
+                           get_cmd=partial(self.get_attribute,
+                                           NIRFSG_ATTR_FREQUENCY),
+                           set_cmd=self.set_frequency,
+                           )
+
+        self.add_parameter(name="power_level",
+                           unit="dBm",
+                           label="power level",
+                           get_cmd=partial(self.get_attribute,
+                                           NIRFSG_ATTR_POWER_LEVEL),
+                           set_cmd=self.set_power_level,
+                           )
+
+        self.add_parameter(name="output_on",
```

Comment:
shall it be renamed to `output_enabled`? that's what the label says, and that's what qcodes also recommends (in its writing drivers page of the docs)

Reply:
Good point. This will require a bit of search-and replace in our scripts but oh well. Fixed in a504142.